spec/models/user_spec.rb

```diff
@@ -0,0 +1,30 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  subject(:users) { User.new(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.') }
+
+  before { subject.save }
+
+  it 'name should be present' do
+    user = User.create(photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
+    expect(user).to_not be_valid
+  end
+
+  it 'post_count should be non negative integer' do
+    subject.posts_counter = -1
+    expect(subject).to_not be_valid
+
+    subject.posts_counter = 0
+    expect(subject).to be_valid
+
+    subject.posts_counter = 5
+    expect(subject).to be_valid
+
+    subject.posts_counter = 'abc'
+    expect(subject).to_not be_valid
+  end
+
+  it 'expect recent posts to be 0 ' do
+    expect(subject.three_recent_posts.length).to eq(0)
+  end
```

Comment:
- [ ] Excellent work you have done creating these tests 👍🏽 . Just a few changes. Kindly wrap the `it` block inside a `describe 'validations'` block. Aside from the fact that it is best practice, using `describe` blocks helps make the structure of your test suite clear. It provides context for the tests and makes it explicit what part of the model you are testing. Your code should look something like this:
```ruby
require 'rails_helper'

RSpec.describe User, type: :model do
  subject(:user) { User.new(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.') }

  before { user.save }

  describe 'validations' do
    it 'name should be present' do
      another_user = User.create(photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
      expect(another_user).to_not be_valid
    end

    it 'post_count should be a non-negative integer' do
      user.posts_counter = -1
      expect(user).to_not be_valid

      user.posts_counter = 0
      expect(user).to be_valid

      user.posts_counter = 5
      expect(user).to be_valid

      user.posts_counter = 'abc'
      expect(user).to_not be_valid
    end
  end

  describe 'recent posts' do
    it 'expect recent posts to be 0' do
      expect(user.three_recent_posts.length).to eq(0)
    end
  end
end